.github/actions/eks-auth/action.yaml

```
@@ -0,0 +1,69 @@
+name: eks-auth
+description: Authenticate to EKS cluster
+
+inputs:
+  aws-account:
+    description: AWS account name to assume role for
+    required: true
+  cluster-name:
+    description: Name of EKS cluster to authenticate to
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - name: Definitions
+      id: defs
+      shell: bash
+      run: |
+        repo="$(basename "${{ github.repository }}" | tr '[:upper:]' '[:lower:]')"
+        echo "session=$repo,${{ github.workflow }},${{ github.run_number }}" >> "$GITHUB_OUTPUT"
+        echo "repository=$repo" >> "$GITHUB_OUTPUT"
+    - name: Assume pr role
+      if: ${{ github.event_name == 'pull_request' }}
+      uses: aws-actions/configure-aws-credentials@v2
+      with:
+        aws-region: eu-west-1
+        role-to-assume: arn:aws:iam::312805905296:role/gha-eks-deployer-pr
+        role-session-name: ${{ steps.defs.outputs.session }}
+        mask-aws-account-id: false
+    - name: Assume push role
+      if: ${{ github.ref == 'refs/heads/main' }}
+      uses: aws-actions/configure-aws-credentials@v2
```

```suggestion
      uses: aws-actions/configure-aws-credentials@main
```
We should to use the latest action instead of the v2, because v2 doesn't allow for the `role-skip-session-tagging` input for example.

---

Yeah it was role chaining, but I believe that got fixed: https://github.com/aws-actions/configure-aws-credentials/blob/v2.2.0/CHANGELOG.md#210-2023-05-31

---

Hm, I remember getting an error yesterday, but I couldn't find that workflow run now.